crates/pg_schema_cache/src/schema_cache.rs

```diff
@@ -42,6 +41,30 @@ impl SchemaCache {
         })
     }
 
+    fn pool_to_conn_str(pool: &PgPool) -> String {
```

Comment:
mhh Connection Str Logic and cache keys on the schema cache feels like a leaking abstraction... why not handling this on the schema cache manager itself? After all, it should manage the schema cache for the workspace and can also be knowledgable of the connection string and handle caching?

Reply:
I had that first, but then I had to maintain another `conn_str: RwLock<String>`, which also felt odd 🤔 I can replicate that in a commit so you can check what that would look like.

My thoughts were that the schema-cache *should* theoretically be the one to know from which DB it pulled the schema. 

Maybe we could form a separate cache-str format for the `SchemaCache`, this doesn't have to be the connection string?

Reply:
yeah, maybe use an inner struct on the manager? This way both the connection string and the data is locked together.

Reply:
done and much better!